homeassistant/core.py

```diff
@@ -252,7 +246,8 @@ def async_block_till_done(self):
 
     def stop(self) -> None:
         """Stop Home Assistant and shuts down all threads."""
-        run_coroutine_threadsafe(self.async_stop(), self.loop)
+        self.loop.call_soon_threadsafe(
+            self.loop.create_task, self.async_stop())
```

Comment:
(note, we include the `run_coroutine_threadsafe` function from Python because it was only introduced in Python 3.5.1